salt/utils/fsutils.py

```diff
@@ -67,9 +67,8 @@ def _blkid_output(out, fs_type=None):
             key, val = items.split('=', 1)
             dev[key.lower()] = val
         if fs_type and dev.get('type', '') == fs_type or not fs_type:
-            if 'type' in dev:
+            if 'type' in dev and fs_type:
```

Comment:
@jfindlay I don't exactly understand what you mean by that, but it was "Always remove "type" key from `dev`, while now it is "Remove "type" key from `dev` only when `fs_type` is defined". Hope this helps.